pkg/glusterfs/controllerserver.go

```diff
@@ -70,85 +69,51 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
-		volSizeBytes = req.GetCapacityRange().GetRequiredBytes()
+	if capRange := req.GetCapacityRange(); capRange != nil {
+		volSizeBytes = capRange.GetRequiredBytes()
 	}
 
 	volSizeMB := int(utils.RoundUpSize(volSizeBytes, 1024*1024))
 
 	// parse the request.
-	parseResp, parseErr := cs.ParseCreateVolRequest(req)
-	if parseErr != nil {
+	parseResp, err := cs.ParseCreateVolRequest(req)
+	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, "failed to parse request")
 	}
 
 	volumeName := parseResp.gdVolReq.Name
 
-	glusterServer, bkpServers, err := cs.checkExistingVolume(volumeName, volSizeMB)
-	if err != nil && err != errVolumeNotFound {
-		return nil, err
-
-	}
-	if err == nil {
-		resp := &csi.CreateVolumeResponse{
-			Volume: &csi.Volume{
-				Id:            volumeName,
-				CapacityBytes: volSizeBytes,
-				Attributes: map[string]string{
-					"glustervol":        volumeName,
-					"glusterserver":     glusterServer,
-					"glusterbkpservers": strings.Join(bkpServers, ":"),
-				},
-			},
-		}
-		return resp, nil
-	}
-
-	// If volume does not exist, provision volume
-	glog.V(4).Infof("Received request to create/provision volume name:%s with size:%d", volumeName, volSizeMB)
-	volMetaMap := make(map[string]string)
-	volMetaMap[volumeOwnerAnn] = glusterfsCSIDriverName
-	volumeReq := api.VolCreateReq{
-		Name:         volumeName,
-		Metadata:     volMetaMap,
-		ReplicaCount: defaultReplicaCount,
-		Size:         uint64(volSizeMB),
-	}
-
-	glog.V(2).Infof("volume request: %+v", volumeReq)
-
-	volumeCreateResp, err := cs.client.VolumeCreate(volumeReq)
+	err = cs.checkExistingVolume(volumeName, volSizeMB)
 	if err != nil {
-		glog.Errorf("failed to create volume : %v", err)
-		return nil, status.Error(codes.Internal, fmt.Sprintf("failed to create volume: %s", err.Error()))
-	}
-
-	glog.V(3).Infof("volume create response : %+v", volumeCreateResp)
-	err = cs.client.VolumeStart(volumeName, true)
-	if err != nil {
-		//we dont need to delete the volume if volume start fails
-		//as we are listing the volumes and starting it again
-		//before sending back the response
-		glog.Errorf("failed to start volume:%v", err)
-		return nil, status.Error(codes.Internal, fmt.Sprintf("failed to start volume %s", err.Error()))
+		if err != errVolumeNotFound {
+			glog.Errorf("Error checking for pre-existing volume: %v", err)
+			return nil, err
+		}
+		// If volume does not exist, provision volume
+		err = cs.doVolumeCreate(volumeName, volSizeMB)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	glusterServer, bkpServers, err = cs.getClusterNodes()
-
+	glusterServer, bkpServers, err := cs.getClusterNodes()
 	if err != nil {
-		glog.Errorf("failed to fetch details of cluster nodes: %v", err)
-		return nil, status.Error(codes.Internal, fmt.Sprintf("error in fecthing peer details %s", err.Error()))
+		glog.Errorf("Failed to get cluster nodes: %v", err)
+		return nil, status.Errorf(codes.Internal, "Failed to get cluster nodes: %v", err)
```

Comment:
returning error message should start with a small case

Reply:
...really? Is that a style thing? That feels so wrong... but if so, I'll do it.

Reply:
Error strings should not start with a capital letter because they'll often be prefixed before printing 
[error-style](https://github.com/golang/go/wiki/Errors#errors)

Reply:
Dang it... all right. :)